pallets/schemas/src/types.rs

```diff
@@ -18,3 +39,56 @@ pub struct SchemaInfo {
 	/// additional control settings for the schema
 	pub settings: SchemaSettings,
 }
+
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
+/// A structure defining name of a schema
+pub struct SchemaName {
+	/// namespace or domain of the schema
+	pub namespace: BoundedVec<u8, ConstU32<NAMESPACE_MAX>>,
+	/// name or descriptor of this schema
+	pub descriptor: BoundedVec<u8, ConstU32<DESCRIPTOR_MAX>>,
+}
+
+impl SchemaName {
+	/// parses and verifies the request and returns the SchemaName type if successful
+	pub fn try_parse<T: Config>(payload: SchemaNamePayload) -> Result<SchemaName, DispatchError> {
+		// check if all ascii
+		let mut str = String::from_utf8(payload.into_inner())
+			.map_err(|_| Error::<T>::InvalidSchemaNameEncoding)?;
+		ensure!(str.is_ascii(), Error::<T>::InvalidSchemaNameEncoding);
+
+		// to canonical form
+		str = str.to_lowercase();
```

Comment:
Question: `lowercase` or `uppercase` for canonical?

Reply:
Should we take advantage of some of the functions developed for the `handles` pallet to do schema name normalization?

Reply:
So I think for schema names based on what it is mentioned in the issue we only support ascii characters. I further simplified the input to only ascii alphanumeric characters but open to changes if somebody feels it's unreasonable

Reply:
some validation steps from handles can be used given the schema creator can add random chars?

Reply:
So we only support `[a-z] [A-Z]` and `-` and `.` in naming so I'm not sure how much the handles sytuff would be helpful but will take a look.

Reply:
I'm closing this comment but upon review if you have specific concern or might have a better way to do something please add comments